subsys/llext/llext_link.c

```diff
@@ -219,6 +220,120 @@ static int llext_lookup_symbol(struct llext *ext, struct llext_loader *ldr, uint
 	return 0;
 }
 
+/**
+ * @brief On RISC-V, PC-relative relocations (PCREL_LO12_I, PCREL_LO12_S) do not refer to
+ * the actual symbol. Instead, they refer to the location of a different instruction in the
+ * same section, which has a PCREL_HI20 relocation. The relocation offset is then computed based
+ * on the location and symbol from the HI20 relocation. 20 bits from the offset go into the
+ * instruction that has the HI20 relocation, and 12 bits go into the PCREL_LO12 instruction.
+ *
+ * @param ext current extension
+ * @param ldr llext loader
+ * @param pcrel_lo12 the elf relocation structure for the PCREL_LO12I/S relocation.
+ * @param shdr ELF section header for the relocation
+ * @param sym ELF symbol for PCREL_LO12I
+ * @param link_addr_out computed link address
+ * @param last_rel_idx index of last relocation entry processed in search
+ *
+ */
+static inline int llext_riscv_find_sym_pcrel(struct llext *ext, struct llext_loader *ldr,
```

Comment:
...and let's remove `inline` - I don't see too many of them used under subsys/llext, the compiler will figure out by itself

Reply:
Done